membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -38,12 +38,37 @@ class AccountController extends LazyLogging {
   lazy val mmaAction = NoCacheAction andThen mmaCorsFilter andThen BackendFromCookieAction
   lazy val mmaCardAction = NoCacheAction andThen corsCardFilter andThen BackendFromCookieAction
 
+
+  def cancelSubscription [P <: SubscriptionPlan.AnyPlan : SubPlanReads] = mmaCardAction.async { implicit request =>
+    val tp = request.touchpoint
+    val cancelForm = Form { single("reason" -> nonEmptyText) }
+
+    val maybeUserId = authenticationService.userId
+    logger.info(s"Attempting to cancel contribution for $maybeUserId")
+
+    (for {
+      user <- EitherT(Future.successful( maybeUserId \/> "no identity cookie for user"))
+      sfUser <- EitherT(tp.contactRepo.get(user).map(_.flatMap(_ \/> s"no SF user $user")))
+      zuoraSubscription <- EitherT(tp.subService.current[P](sfUser).map(_.headOption).map (_ \/> s"no current subscriptions for the sfUser $sfUser"))
```

Comment:
Can we be sure this will always be a contributions subscription ?

Reply:
Yes, we went through the code offline and discussed it.